proto/ocpaths.proto

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ocpaths.proto defines a specification of OpenConfig path support or
+// requirements of a networking entity.
+//
+// This specification is distinct from
+// github.com/openconfig/featureprofiles/proto/feature.proto, which describes
+// the relationship between featureprofiles tests and OC paths and is used to
+// report test coverage of those paths.
+
+syntax = "proto3";
+
+package openconfig.ocpaths;
+
+// OCPaths is the complete list of all OpenConfig paths associated with some
+// entity (e.g. NOS, or path requirements list for a particular device role).
+message OCPaths {
+  // Required OpenConfig paths for this entity.
+  repeated OCPath ocpaths = 1;
+
+  // The semantic version of the OpenConfig public models release
+  // (https://github.com/openconfig/public) associated with the list of
+  // OpenConfig paths.
+  string version = 2;
+}
+
+// OCPath is an individual OpenConfig path path.
+message OCPath {
+  // Flattened OC path without prefixes.
+  string name = 1;
+
+  // Specifies a limited portion of the OC model that a given path is needed to
+  // support.
+  OCPathConstraint ocpath_constraint = 2;
+
+  // Suffix for featureprofiles/feature/<featureprofileid> to which this OCPath
+  // is associated.
+  // This destination featureprofiles folder serves as a grouping of the paths
+  // related to the tests which appear in this folder and subfolders the folder.
+  string featureprofileid = 6;
```

Comment:
Should this be a reference to https://github.com/openconfig/featureprofiles/blob/main/proto/feature.proto#L92 ?

Reply:
If your proposal is to change the type from `string` to `FeatureProfileID`, then the additional benefit would be that a version `optional int32 version` is added. I will argue that this is not necessary:

The purpose of the field `featureprofileid` in OCPath is to record the featureprofile to which the path belongs.
When a feature is generated, and when it is different from a previously-generated feature, then that's when a new version is created.
Therefore, when a new OCPath is created, we don't yet know what feature version should be used, since we don't know when the next feature release is, so this version string would not be useful here.

Added comment to clarify the relationship to `FeatureProfileID`.